composer.json

```
@@ -25,14 +25,9 @@
         }
     },
     "require": {
-        "php": "^7.3 || ~8.0.0",
-        "laminas/laminas-code": "^3.5.0",
-        "laminas/laminas-form": "^2.16.3",
-        "laminas/laminas-i18n": "^2.11.1",
-        "laminas/laminas-zendframework-bridge": "^1.2.0"
-    },
-    "conflict": {
-        "laminas/laminas-code": "<3.5.0 || >=4.0.0"
+        "laminas/laminas-code": "^3.5.1",
+        "laminas/laminas-form": "^2.17.0",
```

I think we can allow version 2 and 3:

```suggestion
        "laminas/laminas-form": "^2.17.0 || ^3.0.1",
```

---

We could, not sure if we want though. Cause then we would keep the dependency to `laminas-code` which is not required with `laminas-form:^3.0` anymore. Additionally, we would still require the conflicts regarding `laminas-code`, which may bring trouble when other libraries are depending on `laminas-code` as well. To me that doesn't look promising and a new major with `laminas-form:^3.0` and without `laminas-code` would be more clean.

---

You are absolutely right! Remove laminas-code and everything works as it should.

---

Good, but that should still be a new major then, right? Cause `laminas-code` has been there before and enables annotation support for `laminas-form:^2.0`. Initially my idea was to suggest a 1.2.0 release with updated depencendies and forms ^2.0 and laminas-code, and a 2.0.0 with form ^3.0 and probably doctrine-annotations as well. Though doctrine-annotations are only required for annotations parsing that would be consistent with the 1.x releaes, which trough including laminas-code enabled annotation parsing right away. In that sense `laminas-mvc-form` would provide `laminas-form` including peer dependencies.

---

@froschdesign push

---

@froschdesign push